polar2grid/utils/legacy_compat.py

```diff
@@ -56,6 +57,10 @@ def convert_old_p2g_date_frmts(frmt):
     return frmt
 
 
+def output_pattern_parser_error(value):
+    raise DeprecationWarning("--output-pattern is deprecated, use --output-filename")
```

Comment:
Interesting. I've never seen warnings used this way before. Usually it is `warnings.warn("some message", DeprecationWarning)`, but maybe in this case this is the right approach? Do the python docs talk about this?

Reply:
Looking at the docs, it probably shouldn't be used that way.  However, it seems silly to keep moving forward.  However, the code will raise an argument parse exception later, so it is proabably fine just to use warnings.warn as was intended with DeprecationWarning (not DeprecationError).

Reply:
Note warnings.warn does not print out DeprecationWarnings by default. Use the default UserWarning if you want that.